src/main/java/org/odessajavaclub/category/adapter/in/web/CategoryController.java

```diff
@@ -15,28 +19,38 @@
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
-import java.util.List;
-
-import static java.util.stream.Collectors.toList;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @RestController
 @RequiredArgsConstructor
 @RequestMapping(value = "/category")
 public class CategoryController {
 
+    @Value("${spring.data.web.pageable.default-page-size}")
+    private int defaultPageSize;
+
     private final GetCategoriesQuery getCategoriesQuery;
     private final GetCategoryQuery getCategoryQuery;
     private final CreateCategoryUseCase createCategoryUseCase;
     private final UpdateCategoryUseCase updateCategoryUseCase;
     private final DeleteCategoryUseCase deleteCategoryUseCase;
 
-    @GetMapping("/list")
-    public List<CategoryDto> getAll() {
-        return getCategoriesQuery.getCategories().stream()
+    @GetMapping
+    public Page<CategoryDto> getCategories(@RequestParam(required = false) Integer page,
```

Comment:
You can use smth like this instead of using Optional:
```
@RequestParam(required = false, default = "0") int page,
@RequestParam(required = false, default = "${spring.data.web.pageable.default-page-size}" int size
```